File: Data Structures & Algorithms/sort-an-array/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    sortArray(nums) {
        let maxNum = 0
        for ( const num of nums) {
            if ( num > maxNum) {
                maxNum = num
            }
        }

        const maxDigit = `${maxNum}`.length
        console.log(maxDigit)

        let collected = nums
        for ( let currentDigit = 0; currentDigit < maxDigit; currentDigit++) {
            const buckets = [[], [], [], [], [], [], [], [], [], [], []]

            for ( const num of collected) {
                const index = Math.floor(num / (10 **currentDigit)) % 10
                buckets[index].push(num)
            }

            collected = []
            for ( const bucket of buckets) {
                for ( const value of bucket) {
                    collected.push(value)
                }
            }

            console.log(collected)
        }

        return collected
    }
}

File: Data Structures & Algorithms/jump-game-ii/submission-0.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        minjump = [ math.inf for num in nums]
        minjump[-1] = 0

        for i in range(len(nums) - 2, -1, -1):
            for j in range(i + 1, i + nums[i] + 1):
                minjump[i] = min(minjump[i], minjump[j] + 1)

        print(minjump)
        return minjump[0]